module/events/finalization_actor.go

```diff
@@ -0,0 +1,70 @@
+package events
+
+import (
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/tracker"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+)
+
+// OnBlockFinalized is invoked when a new block is finalized. It is possible that
+// blocks will be skipped.
+type OnBlockFinalized func(block *model.Block) error
+
+// FinalizationActor is an event responder worker which can be embedded in a component
+// to simplify the plumbing required to respond to block finalization events.
+// This worker is designed to respond to a newly finalized blocks on a best-effort basis,
+// meaning that it may skip blocks when finalization occurs more quickly.
+// CAUTION: This is suitable for use only when the handler can tolerate skipped blocks.
+type FinalizationActor struct {
+	newestFinalized *tracker.NewestBlockTracker
+	notifier        engine.Notifier
+	handler         OnBlockFinalized
+}
+
+// NewFinalizationActor creates a new FinalizationActor, and returns the worker routine
+// and event consumer required to operate it.
+// The caller MUST:
+//   - start the returned component.ComponentWorker function
+//   - subscribe the returned FinalizationActor to OnBlockFinalized events
+func NewFinalizationActor(handler OnBlockFinalized) (*FinalizationActor, component.ComponentWorker) {
+	actor := &FinalizationActor{
+		newestFinalized: tracker.NewNewestBlockTracker(),
+		notifier:        engine.NewNotifier(),
+		handler:         handler,
+	}
+	return actor, actor.worker
+}
+
+// worker is the worker function exposed by the FinalizationActor. It should be
+// attached to a ComponentBuilder by the higher-level component using CreateWorker.
+// It processes each new finalized block by invoking the OnBlockFinalized callback.
+func (actor *FinalizationActor) worker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
```

Comment:
can we rename this to `workerLogic` or similar? I feel that the naming of `ComponentWorker` is misleading, because the thing is _not_ a worker. It is the logic to be _executed by_ a worker ... but the latter is out of scope for this PR

Reply:
The difference between the function describing a worker's logic and the worker itself, in the context of the `ComponentManager`, is essentially whether it's invoked as a goroutine. The "invoke as a goroutine" part is entirely managed within the scope of the `ComponentManager`; we don't really see types that have both `func workerLogic(...)` _and_ `func worker(...) { go workerLogic(...) }`, so I think the opportunity for confusion here is pretty small. 

I did change the name here, but FWIW I personally think the scheme of naming functions which will be invoked as workers `XYZWorker`, and the `ComponentWorker` type name, makes sense.